### PR TITLE
Enable ExportTimeSeriesLog to export partial time series.

### LINKED ISSUE #14545: Enable ExportTimeSeriesLog to export time with various unit
Enable ExportTimeSeriesLog to export time with various unit such as second.  

Enable ExportTimeSeriesLog to export time as absolute time. 

Enable ExportTimeSeriesLog to export partial series within a time range.

### PR DESCRIPTION
**Pull request by wdzhou**
Fixes #14545.

### CODE REVIEW

**Comment by wdzhou** [comment]
For tester:

Here is a script to verify the new feature:
```GenerateEventsFilter(InputWorkspace='VULCAN_71087_event', OutputWorkspace='splittermatrix', InformationWorkspace='infows', 
                                   FastLog=True, StartTime='1', StopTime='6', TimeInterval='1', MinimumLogValue=0, MaximumLogValue=2, LogValueInterval=1, TitleOfSplitters='aa')
FilterEvents(InputWorkspace='VULCAN_71087_event', SplitterWorkspace='splittermatrix', OutputWorkspaceBaseName='Result3', InformationWorkspace='infows', 
                   GroupWorkspaces=True, OutputWorkspaceIndexedFrom1=True, SpectrumWithoutDetector='Skip only if TOF correction')

# Generate a matrix splitters with relative time
import numpy
vecx = numpy.array([1, 2, 3, 4, 5, 6])
vecx *= 1.E9
vecy = [0, 1, 2, 3, 4]
vece = [1, 1, 1, 1, 1]

matrixsplitterws = CreateWorkspace(DataX=vecx, DataY=vecy, DataE=vece, NSpec=1)
    
FilterEvents(InputWorkspace='VULCAN_71087_event', SplitterWorkspace=matrixsplitterws, 
                   OutputWorkspaceBaseName='Result2',
                   GroupWorkspaces=True, OutputWorkspaceIndexedFrom1=True, 
                   SpectrumWithoutDetector='Skip only if TOF correction',
                   RelativeTime=True)
           
                   
for iws in xrange(1, 6):
    ws1 = mtd['Result2_%d' % iws]
    ws2 = mtd['Result3_%d' % iws]
    print 'output workspace', iws, 'diff in number of events = ', ws1.getNumberEvents() - ws2.getNumberEvents()


FilterEvents(InputWorkspace='VULCAN_71087_event', SplitterWorkspace=matrixsplitterws, 
                   OutputWorkspaceBaseName='Result1',
                   GroupWorkspaces=True, OutputWorkspaceIndexedFrom1=True, 
                   SpectrumWithoutDetector='Skip only if TOF correction',
                   FilterStartTime='2015-06-23T19:39:56.652849666',
                   RelativeTime=True)

for iws in xrange(1, 6):
    ws1 = mtd['Result1_%d' % iws]
    ws2 = mtd['Result3_%d' % iws]
    print 'output workspace', iws, 'diff in number of events = ', ws1.getNumberEvents() - ws2.getNumberEvents()
```

File: Framework/Algorithms/src/ExportTimeSeriesLog.cpp [hunk 1]
```diff
@@ -65,33 +88,52 @@ void ExportTimeSeriesLog::init() {
...
 
   return;
 }
-
+/*
+ *  * @param logname ::
+  * @param numentries ::
+  * @param outputeventws ::
+ * true.
+ */
```
Comment (FedeMPouzols) [inline Framework/Algorithms/src/ExportTimeSeriesLog.cpp]: This looks like a leftover that could be removed?

**Comment by FedeMPouzols** [comment]
Code looks good and it comes with new tests and documentation.

@wdzhou : have you added a release note about this? A good place for it would be the [improved algorithms](http://www.mantidproject.org/ReleaseNotes_3_6_Framework_Changes#Improved).

**Comment by FedeMPouzols** [comment]
@wdzhou : I'm going to merge this so it is included in the beta testing versions. Please remember to check the point about the release notes.

**Comment by wdzhou** [comment]
@FedeMPouzols Thanks for closing this.  I've made the change in the release note.

**Pull request merged**